dhis2/src/main/resources/webapp/js/controllers.js

```diff
@@ -35,6 +35,13 @@
             };
 
             $scope.submit = function () {
+                if(!$scope.settings.serverURI.match("^https?:\/\/.+")) {
+                    if($scope.settings.serverURI.match("^.+:\/\/.+")) {
```

Comment:
So what are we doing here? If I enter ftp:// for example, it replaces it with http? This is a feature we do not really need, all we need is to check if this is false, if false add the http:// protocol.